tools/autograd/gen_variable_type.py

```diff
@@ -418,380 +319,184 @@ def emit_record_trace(env, declaration):
         local['trace_name'] = declaration['api_name']
         if local['trace_name'].endswith('_'):
             local['trace_name'] = local['trace_name'][:-1]
+        local['trace_outputs'] = get_trace_outputs(declaration)
 
         combined = nested_dict(local, nested_dict(env, declaration))
         return RECORD_TRACE.substitute(combined)
 
-    def emit_check_no_requires_grad(tensor_args, args_with_derivatives):
-        """Checks that arguments without derivatives don't require grad"""
-        body = []
-        for arg in tensor_args:
-            name = arg['name']
-            if name in args_with_derivatives:
-                continue
-            if name == 'output':
-                # Double-backwards definitions sometimes take in 'input' and
-                # 'output', but only define the derivative for input.
-                continue
-            if arg['dynamic_type'] in {'IndexTensor', 'BoolTensor'}:
-                continue
-            body.append('check_no_requires_grad({}, "{}");'.format(name, name))
-        return body
-
-    def emit_body_via_type(declaration):
-        env = {}
-        body = []
-
-        combined = nested_dict(env, declaration)
-        arguments = declaration['arguments']
-
-        tensor_args = [arg for arg in arguments if arg['simple_type'] in {'Tensor', 'TensorList'}]
-        env['tensor_args'] = [arg['name'] for arg in tensor_args]
-
-        if declaration['inplace']:
-            env['return_value'] = 'self'
-            env['trace_outputs'] = '{ self }'
-        elif declaration['return_type'] == 'std::vector<Tensor>':
-            env['return_value'] = 'ret'
-            env['trace_outputs'] = 'cast_tensor_list(ret)'
-        else:
-            env['return_value'] = '{}(std::move(ret))'.format(declaration['return_type'])
-            env['trace_outputs'] = get_trace_outputs(declaration)
-
-        if declaration['name'] in DONT_RECORD_TRACE:
-            env['record_trace'] = ''
+    def declare_return_values():
```

Comment:
nit: I feel like "values" isn't the right name here, maybe "objects"?  When I think of return values I think of e.g. "5" which you don't declare.

Reply:
Changed to "declare_returned_variables"